amm/runtime/src/main/scala/ammonite/runtime/tools/IvyThing.scala

```diff
@@ -62,14 +62,15 @@ object IvyThing{
           val loadedArtifacts = scalaz.concurrent.Task.gatherUnordered(
             for (a <- artifacts)
               yield coursier.Cache.file(a, logger = logger).run
+                .map(a.isOptional -> _)
           ).run
 
-          val errors = loadedArtifacts.collect { case scalaz.-\/(x) => x }
-          val successes = loadedArtifacts.collect { case scalaz.\/-(x) => x }
+          val errors = loadedArtifacts.collect { case (false, scalaz.-\/(x)) => x }
```

Comment:
Only not found errors should be ignored, the others should be retained (and should make the resolution fail overall). `coursier.FileError` have [a method](https://github.com/coursier/coursier/blob/b21042a42a6b2718ec1116f28f76dd769112c6aa/cache/src/main/scala/coursier/FileError.scala#L11) indicating whether an error corresponds to not found.